src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs

```diff
@@ -17,7 +17,12 @@ public static class DatasetUtils
         /// Downloads the housing dataset from the ML.NET repo.
         /// </summary>
         public static string DownloadHousingRegressionDataset()
-        => Download("https://raw.githubusercontent.com/dotnet/machinelearning/024bd4452e1d3660214c757237a19d6123f951ca/test/data/housing.txt", "housing.txt");
+        {
```

Comment:
Is this change necessary in this PR?

I don't think our tests should be calling this method - BTW

Reply:
This change is necessary if we want to use `LoadHousingRegressionDataset` in our tests because there is a race condition on the file lock, so tests will sometimes fail.

Can you explain a bit more why you don't want to use this in tests? Is it that we don't want to use the SamplesUtils project in Tests, or that we shouldn't be downloading data for tests?

If it's the former, check out issue #2420 . We're going to make this a standalone Datasets/ (or some such name) outside of the NuGet project to use in Samples and Tests.

If it's the latter, we are already downloading datasets for tests. But now that I mention it, we can actually add an optional input to LoadHousingRegressionDataset and friends that can load the file from the tests/data/ directory. I'll add this capability now.

---
In reply to: [255236393](https://github.com/dotnet/machinelearning/pull/2470#discussion_r255236393) [](ancestors = 255236393)

Reply:
Added.

---
In reply to: [255256550](https://github.com/dotnet/machinelearning/pull/2470#discussion_r255256550) [](ancestors = 255256550,255236393)

Reply:
There are 2 datasets our tests should use.

1. Datasets checked into `test\data`.
2. Datasets that are downloaded into `test\data\external` through the [`DownloadExternalTestFiles`](https://github.com/dotnet/machinelearning/blob/a254bf58c47c1a00348fcc628f48e26dca959ee6/build.proj#L88-L93) build step.

We shouldn't have the test code be downloading random things.

Reply:
Got it. This has been updated to use the local dataset in the `test\data` folder. I'll chase down any other tests using these `Download` commands as I migrate API-Scenario tests to Functional.Tests/

---
In reply to: [255264819](https://github.com/dotnet/machinelearning/pull/2470#discussion_r255264819) [](ancestors = 255264819)